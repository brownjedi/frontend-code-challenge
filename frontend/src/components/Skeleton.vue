<template>
  <div :class="styles.skeleton"></div>
</template>

<script lang="ts">
import { defineComponent, useCssModule } from 'vue'

export default defineComponent({
  setup() {
    const styles = useCssModule()

    return { styles }
  },
})
</script>

<style lang="scss" module>
/* The loading Class */
.skeleton {
  position: relative;
  background-color: #e2e2e2;
  width: 100%;
  height: 100%;
}

/* The moving element */
.skeleton::after {
  display: block;
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background: -webkit-gradient(
    linear,
    left top,
    right top,
    from(transparent),
    color-stop(rgba(255, 255, 255, 0.2)),
    to(transparent)
  );

  background: linear-gradient(90deg, transparent, rgba(255, 255, 255, 0.2), transparent);

  /* Adding animation */
  animation: loading 1.2s infinite;
}

/* Loading Animation */
@keyframes loading {
  0% {
    background-position: -200px 0;
  }
  100% {
    background-position: calc(200px + 100%) 0;
  }
}
</style>
