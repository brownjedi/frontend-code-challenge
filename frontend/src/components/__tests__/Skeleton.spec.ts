import { shallowMount } from '@vue/test-utils'
import Skeleton from '@/components/Skeleton.vue'

describe('components/Skeleton.vue', () => {
  it('has the expected html structure', () => {
    const wrapper = shallowMount(Skeleton)
    expect(wrapper.element).toMatchSnapshot()
  })
})
