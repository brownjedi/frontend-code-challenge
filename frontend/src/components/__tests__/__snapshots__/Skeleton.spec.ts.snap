// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Skeleton.vue has the expected html structure 1`] = `
<div
  class="skeleton"
/>
`;
